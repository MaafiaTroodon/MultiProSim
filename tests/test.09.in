6 5 3
Proc1 7 1 1
LOOP 10
  DOOP 1
  SEND 201
  RECV 301
  BLOCK 3
END
HALT

Proc2 7 1 2
LOOP 10
  DOOP 3
  RECV 101
  SEND 301
  BLOCK 5
END
HALT

Proc3 7 1 3
LOOP 10
  DOOP 4
  RECV 201
  SEND 101
  BLOCK 2
END
HALT

Proc4 7 1 1
LOOP 10
  DOOP 1
  SEND 302
  RECV 202
  BLOCK 3
END
HALT

Proc5 7 1 2
LOOP 10
  DOOP 3
  RECV 302
  SEND 102
  BLOCK 5
END
HALT

Proc6 7 1 3
LOOP 10
  DOOP 4
  RECV 102
  SEND 202
  BLOCK 2
END
HALT
