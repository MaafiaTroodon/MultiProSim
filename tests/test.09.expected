[01] 00000: process 1 new
[01] 00000: process 1 ready
[01] 00000: process 1 running
[01] 00000: process 2 new
[01] 00000: process 2 ready
[01] 00001: process 1 ready
[01] 00001: process 2 running
[01] 00002: process 1 running
[01] 00002: process 2 ready
[01] 00003: process 1 blocked (send)
[01] 00003: process 2 running
[01] 00004: process 2 blocked (send)
[01] 00008: process 1 ready
[01] 00008: process 1 running
[01] 00009: process 1 blocked (recv)
[01] 00011: process 2 ready
[01] 00011: process 2 running
[01] 00012: process 1 blocked
[01] 00012: process 2 blocked (recv)
[01] 00015: process 1 ready
[01] 00015: process 1 running
[01] 00015: process 2 blocked
[01] 00016: process 1 ready
[01] 00016: process 1 running
[01] 00017: process 1 blocked (send)
[01] 00018: process 2 ready
[01] 00018: process 2 running
[01] 00019: process 2 ready
[01] 00019: process 2 running
[01] 00020: process 1 ready
[01] 00020: process 1 running
[01] 00020: process 2 blocked (send)
[01] 00021: process 1 blocked (recv)
[01] 00025: process 2 ready
[01] 00025: process 2 running
[01] 00026: process 1 blocked
[01] 00026: process 2 blocked (recv)
[01] 00029: process 1 ready
[01] 00029: process 1 running
[01] 00029: process 2 blocked
[01] 00030: process 1 ready
[01] 00030: process 1 running
[01] 00031: process 1 blocked (send)
[01] 00032: process 2 ready
[01] 00032: process 2 running
[01] 00033: process 2 ready
[01] 00033: process 2 running
[01] 00034: process 1 ready
[01] 00034: process 1 running
[01] 00034: process 2 blocked (send)
[01] 00035: process 1 blocked (recv)
[01] 00039: process 2 ready
[01] 00039: process 2 running
[01] 00040: process 1 blocked
[01] 00040: process 2 blocked (recv)
[01] 00043: process 1 ready
[01] 00043: process 1 running
[01] 00043: process 2 blocked
[01] 00044: process 1 ready
[01] 00044: process 1 running
[01] 00045: process 1 blocked (send)
[01] 00046: process 2 ready
[01] 00046: process 2 running
[01] 00047: process 2 ready
[01] 00047: process 2 running
[01] 00048: process 1 ready
[01] 00048: process 1 running
[01] 00048: process 2 blocked (send)
[01] 00049: process 1 blocked (recv)
[01] 00053: process 2 ready
[01] 00053: process 2 running
[01] 00054: process 1 blocked
[01] 00054: process 2 blocked (recv)
[01] 00057: process 1 ready
[01] 00057: process 1 running
[01] 00057: process 2 blocked
[01] 00058: process 1 ready
[01] 00058: process 1 running
[01] 00059: process 1 blocked (send)
[01] 00060: process 2 ready
[01] 00060: process 2 running
[01] 00061: process 2 ready
[01] 00061: process 2 running
[01] 00062: process 1 ready
[01] 00062: process 1 running
[01] 00062: process 2 blocked (send)
[01] 00063: process 1 blocked (recv)
[01] 00067: process 2 ready
[01] 00067: process 2 running
[01] 00068: process 1 blocked
[01] 00068: process 2 blocked (recv)
[01] 00071: process 1 ready
[01] 00071: process 1 running
[01] 00071: process 2 blocked
[01] 00072: process 1 ready
[01] 00072: process 1 running
[01] 00073: process 1 blocked (send)
[01] 00074: process 2 ready
[01] 00074: process 2 running
[01] 00075: process 2 ready
[01] 00075: process 2 running
[01] 00076: process 1 ready
[01] 00076: process 1 running
[01] 00076: process 2 blocked (send)
[01] 00077: process 1 blocked (recv)
[01] 00081: process 2 ready
[01] 00081: process 2 running
[01] 00082: process 1 blocked
[01] 00082: process 2 blocked (recv)
[01] 00085: process 1 ready
[01] 00085: process 1 running
[01] 00085: process 2 blocked
[01] 00086: process 1 ready
[01] 00086: process 1 running
[01] 00087: process 1 blocked (send)
[01] 00088: process 2 ready
[01] 00088: process 2 running
[01] 00089: process 2 ready
[01] 00089: process 2 running
[01] 00090: process 1 ready
[01] 00090: process 1 running
[01] 00090: process 2 blocked (send)
[01] 00091: process 1 blocked (recv)
[01] 00095: process 2 ready
[01] 00095: process 2 running
[01] 00096: process 1 blocked
[01] 00096: process 2 blocked (recv)
[01] 00099: process 1 ready
[01] 00099: process 1 running
[01] 00099: process 2 blocked
[01] 00100: process 1 ready
[01] 00100: process 1 running
[01] 00101: process 1 blocked (send)
[01] 00102: process 2 ready
[01] 00102: process 2 running
[01] 00103: process 2 ready
[01] 00103: process 2 running
[01] 00104: process 1 ready
[01] 00104: process 1 running
[01] 00104: process 2 blocked (send)
[01] 00105: process 1 blocked (recv)
[01] 00109: process 2 ready
[01] 00109: process 2 running
[01] 00110: process 1 blocked
[01] 00110: process 2 blocked (recv)
[01] 00113: process 1 ready
[01] 00113: process 1 running
[01] 00113: process 2 blocked
[01] 00114: process 1 ready
[01] 00114: process 1 running
[01] 00115: process 1 blocked (send)
[01] 00116: process 2 ready
[01] 00116: process 2 running
[01] 00117: process 2 ready
[01] 00117: process 2 running
[01] 00118: process 1 ready
[01] 00118: process 1 running
[01] 00118: process 2 blocked (send)
[01] 00119: process 1 blocked (recv)
[01] 00123: process 2 ready
[01] 00123: process 2 running
[01] 00124: process 1 blocked
[01] 00124: process 2 blocked (recv)
[01] 00127: process 1 ready
[01] 00127: process 1 running
[01] 00127: process 2 blocked
[01] 00128: process 1 ready
[01] 00128: process 1 running
[01] 00129: process 1 blocked (send)
[01] 00130: process 2 ready
[01] 00130: process 2 running
[01] 00131: process 2 ready
[01] 00131: process 2 running
[01] 00132: process 1 ready
[01] 00132: process 1 running
[01] 00132: process 2 blocked (send)
[01] 00133: process 1 blocked (recv)
[01] 00137: process 2 ready
[01] 00137: process 2 running
[01] 00138: process 1 blocked
[01] 00138: process 2 blocked (recv)
[01] 00141: process 1 finished
[01] 00141: process 2 blocked
[01] 00144: process 2 finished
[02] 00000: process 1 new
[02] 00000: process 1 ready
[02] 00000: process 1 running
[02] 00000: process 2 new
[02] 00000: process 2 ready
[02] 00003: process 1 ready
[02] 00003: process 2 running
[02] 00006: process 1 running
[02] 00006: process 2 ready
[02] 00007: process 1 blocked (recv)
[02] 00007: process 2 running
[02] 00008: process 1 ready
[02] 00008: process 1 running
[02] 00008: process 2 blocked (recv)
[02] 00009: process 1 blocked (send)
[02] 00010: process 1 blocked
[02] 00013: process 2 ready
[02] 00013: process 2 running
[02] 00014: process 2 blocked (send)
[02] 00015: process 1 ready
[02] 00015: process 1 running
[02] 00015: process 2 blocked
[02] 00018: process 1 ready
[02] 00018: process 1 running
[02] 00019: process 1 blocked (recv)
[02] 00020: process 1 ready
[02] 00020: process 1 running
[02] 00020: process 2 ready
[02] 00021: process 1 blocked (send)
[02] 00021: process 2 running
[02] 00024: process 1 blocked
[02] 00024: process 2 ready
[02] 00024: process 2 running
[02] 00025: process 2 blocked (recv)
[02] 00027: process 2 ready
[02] 00027: process 2 running
[02] 00028: process 2 blocked (send)
[02] 00029: process 1 ready
[02] 00029: process 1 running
[02] 00029: process 2 blocked
[02] 00032: process 1 ready
[02] 00032: process 1 running
[02] 00033: process 1 blocked (recv)
[02] 00034: process 1 ready
[02] 00034: process 1 running
[02] 00034: process 2 ready
[02] 00035: process 1 blocked (send)
[02] 00035: process 2 running
[02] 00038: process 1 blocked
[02] 00038: process 2 ready
[02] 00038: process 2 running
[02] 00039: process 2 blocked (recv)
[02] 00041: process 2 ready
[02] 00041: process 2 running
[02] 00042: process 2 blocked (send)
[02] 00043: process 1 ready
[02] 00043: process 1 running
[02] 00043: process 2 blocked
[02] 00046: process 1 ready
[02] 00046: process 1 running
[02] 00047: process 1 blocked (recv)
[02] 00048: process 1 ready
[02] 00048: process 1 running
[02] 00048: process 2 ready
[02] 00049: process 1 blocked (send)
[02] 00049: process 2 running
[02] 00052: process 1 blocked
[02] 00052: process 2 ready
[02] 00052: process 2 running
[02] 00053: process 2 blocked (recv)
[02] 00055: process 2 ready
[02] 00055: process 2 running
[02] 00056: process 2 blocked (send)
[02] 00057: process 1 ready
[02] 00057: process 1 running
[02] 00057: process 2 blocked
[02] 00060: process 1 ready
[02] 00060: process 1 running
[02] 00061: process 1 blocked (recv)
[02] 00062: process 1 ready
[02] 00062: process 1 running
[02] 00062: process 2 ready
[02] 00063: process 1 blocked (send)
[02] 00063: process 2 running
[02] 00066: process 1 blocked
[02] 00066: process 2 ready
[02] 00066: process 2 running
[02] 00067: process 2 blocked (recv)
[02] 00069: process 2 ready
[02] 00069: process 2 running
[02] 00070: process 2 blocked (send)
[02] 00071: process 1 ready
[02] 00071: process 1 running
[02] 00071: process 2 blocked
[02] 00074: process 1 ready
[02] 00074: process 1 running
[02] 00075: process 1 blocked (recv)
[02] 00076: process 1 ready
[02] 00076: process 1 running
[02] 00076: process 2 ready
[02] 00077: process 1 blocked (send)
[02] 00077: process 2 running
[02] 00080: process 1 blocked
[02] 00080: process 2 ready
[02] 00080: process 2 running
[02] 00081: process 2 blocked (recv)
[02] 00083: process 2 ready
[02] 00083: process 2 running
[02] 00084: process 2 blocked (send)
[02] 00085: process 1 ready
[02] 00085: process 1 running
[02] 00085: process 2 blocked
[02] 00088: process 1 ready
[02] 00088: process 1 running
[02] 00089: process 1 blocked (recv)
[02] 00090: process 1 ready
[02] 00090: process 1 running
[02] 00090: process 2 ready
[02] 00091: process 1 blocked (send)
[02] 00091: process 2 running
[02] 00094: process 1 blocked
[02] 00094: process 2 ready
[02] 00094: process 2 running
[02] 00095: process 2 blocked (recv)
[02] 00097: process 2 ready
[02] 00097: process 2 running
[02] 00098: process 2 blocked (send)
[02] 00099: process 1 ready
[02] 00099: process 1 running
[02] 00099: process 2 blocked
[02] 00102: process 1 ready
[02] 00102: process 1 running
[02] 00103: process 1 blocked (recv)
[02] 00104: process 1 ready
[02] 00104: process 1 running
[02] 00104: process 2 ready
[02] 00105: process 1 blocked (send)
[02] 00105: process 2 running
[02] 00108: process 1 blocked
[02] 00108: process 2 ready
[02] 00108: process 2 running
[02] 00109: process 2 blocked (recv)
[02] 00111: process 2 ready
[02] 00111: process 2 running
[02] 00112: process 2 blocked (send)
[02] 00113: process 1 ready
[02] 00113: process 1 running
[02] 00113: process 2 blocked
[02] 00116: process 1 ready
[02] 00116: process 1 running
[02] 00117: process 1 blocked (recv)
[02] 00118: process 1 ready
[02] 00118: process 1 running
[02] 00118: process 2 ready
[02] 00119: process 1 blocked (send)
[02] 00119: process 2 running
[02] 00122: process 1 blocked
[02] 00122: process 2 ready
[02] 00122: process 2 running
[02] 00123: process 2 blocked (recv)
[02] 00125: process 2 ready
[02] 00125: process 2 running
[02] 00126: process 2 blocked (send)
[02] 00127: process 1 ready
[02] 00127: process 1 running
[02] 00127: process 2 blocked
[02] 00130: process 1 ready
[02] 00130: process 1 running
[02] 00131: process 1 blocked (recv)
[02] 00132: process 1 ready
[02] 00132: process 1 running
[02] 00132: process 2 ready
[02] 00133: process 1 blocked (send)
[02] 00133: process 2 running
[02] 00136: process 1 blocked
[02] 00136: process 2 ready
[02] 00136: process 2 running
[02] 00137: process 2 blocked (recv)
[02] 00139: process 2 ready
[02] 00139: process 2 running
[02] 00140: process 2 blocked (send)
[02] 00141: process 1 finished
[02] 00141: process 2 blocked
[02] 00146: process 2 finished
[03] 00000: process 1 new
[03] 00000: process 1 ready
[03] 00000: process 1 running
[03] 00000: process 2 new
[03] 00000: process 2 ready
[03] 00004: process 1 ready
[03] 00004: process 2 running
[03] 00008: process 1 running
[03] 00008: process 2 ready
[03] 00009: process 1 blocked (recv)
[03] 00009: process 2 running
[03] 00010: process 1 ready
[03] 00010: process 1 running
[03] 00010: process 2 blocked (recv)
[03] 00011: process 1 blocked (send)
[03] 00011: process 2 ready
[03] 00011: process 2 running
[03] 00012: process 1 blocked
[03] 00012: process 2 blocked (send)
[03] 00013: process 2 blocked
[03] 00014: process 1 ready
[03] 00014: process 1 running
[03] 00015: process 2 ready
[03] 00018: process 1 ready
[03] 00018: process 2 running
[03] 00022: process 1 running
[03] 00022: process 2 ready
[03] 00023: process 1 blocked (recv)
[03] 00023: process 2 running
[03] 00024: process 1 ready
[03] 00024: process 1 running
[03] 00024: process 2 blocked (recv)
[03] 00025: process 1 blocked (send)
[03] 00025: process 2 ready
[03] 00025: process 2 running
[03] 00026: process 1 blocked
[03] 00026: process 2 blocked (send)
[03] 00027: process 2 blocked
[03] 00028: process 1 ready
[03] 00028: process 1 running
[03] 00029: process 2 ready
[03] 00032: process 1 ready
[03] 00032: process 2 running
[03] 00036: process 1 running
[03] 00036: process 2 ready
[03] 00037: process 1 blocked (recv)
[03] 00037: process 2 running
[03] 00038: process 1 ready
[03] 00038: process 1 running
[03] 00038: process 2 blocked (recv)
[03] 00039: process 1 blocked (send)
[03] 00039: process 2 ready
[03] 00039: process 2 running
[03] 00040: process 1 blocked
[03] 00040: process 2 blocked (send)
[03] 00041: process 2 blocked
[03] 00042: process 1 ready
[03] 00042: process 1 running
[03] 00043: process 2 ready
[03] 00046: process 1 ready
[03] 00046: process 2 running
[03] 00050: process 1 running
[03] 00050: process 2 ready
[03] 00051: process 1 blocked (recv)
[03] 00051: process 2 running
[03] 00052: process 1 ready
[03] 00052: process 1 running
[03] 00052: process 2 blocked (recv)
[03] 00053: process 1 blocked (send)
[03] 00053: process 2 ready
[03] 00053: process 2 running
[03] 00054: process 1 blocked
[03] 00054: process 2 blocked (send)
[03] 00055: process 2 blocked
[03] 00056: process 1 ready
[03] 00056: process 1 running
[03] 00057: process 2 ready
[03] 00060: process 1 ready
[03] 00060: process 2 running
[03] 00064: process 1 running
[03] 00064: process 2 ready
[03] 00065: process 1 blocked (recv)
[03] 00065: process 2 running
[03] 00066: process 1 ready
[03] 00066: process 1 running
[03] 00066: process 2 blocked (recv)
[03] 00067: process 1 blocked (send)
[03] 00067: process 2 ready
[03] 00067: process 2 running
[03] 00068: process 1 blocked
[03] 00068: process 2 blocked (send)
[03] 00069: process 2 blocked
[03] 00070: process 1 ready
[03] 00070: process 1 running
[03] 00071: process 2 ready
[03] 00074: process 1 ready
[03] 00074: process 2 running
[03] 00078: process 1 running
[03] 00078: process 2 ready
[03] 00079: process 1 blocked (recv)
[03] 00079: process 2 running
[03] 00080: process 1 ready
[03] 00080: process 1 running
[03] 00080: process 2 blocked (recv)
[03] 00081: process 1 blocked (send)
[03] 00081: process 2 ready
[03] 00081: process 2 running
[03] 00082: process 1 blocked
[03] 00082: process 2 blocked (send)
[03] 00083: process 2 blocked
[03] 00084: process 1 ready
[03] 00084: process 1 running
[03] 00085: process 2 ready
[03] 00088: process 1 ready
[03] 00088: process 2 running
[03] 00092: process 1 running
[03] 00092: process 2 ready
[03] 00093: process 1 blocked (recv)
[03] 00093: process 2 running
[03] 00094: process 1 ready
[03] 00094: process 1 running
[03] 00094: process 2 blocked (recv)
[03] 00095: process 1 blocked (send)
[03] 00095: process 2 ready
[03] 00095: process 2 running
[03] 00096: process 1 blocked
[03] 00096: process 2 blocked (send)
[03] 00097: process 2 blocked
[03] 00098: process 1 ready
[03] 00098: process 1 running
[03] 00099: process 2 ready
[03] 00102: process 1 ready
[03] 00102: process 2 running
[03] 00106: process 1 running
[03] 00106: process 2 ready
[03] 00107: process 1 blocked (recv)
[03] 00107: process 2 running
[03] 00108: process 1 ready
[03] 00108: process 1 running
[03] 00108: process 2 blocked (recv)
[03] 00109: process 1 blocked (send)
[03] 00109: process 2 ready
[03] 00109: process 2 running
[03] 00110: process 1 blocked
[03] 00110: process 2 blocked (send)
[03] 00111: process 2 blocked
[03] 00112: process 1 ready
[03] 00112: process 1 running
[03] 00113: process 2 ready
[03] 00116: process 1 ready
[03] 00116: process 2 running
[03] 00120: process 1 running
[03] 00120: process 2 ready
[03] 00121: process 1 blocked (recv)
[03] 00121: process 2 running
[03] 00122: process 1 ready
[03] 00122: process 1 running
[03] 00122: process 2 blocked (recv)
[03] 00123: process 1 blocked (send)
[03] 00123: process 2 ready
[03] 00123: process 2 running
[03] 00124: process 1 blocked
[03] 00124: process 2 blocked (send)
[03] 00125: process 2 blocked
[03] 00126: process 1 ready
[03] 00126: process 1 running
[03] 00127: process 2 ready
[03] 00130: process 1 ready
[03] 00130: process 2 running
[03] 00134: process 1 running
[03] 00134: process 2 ready
[03] 00135: process 1 blocked (recv)
[03] 00135: process 2 running
[03] 00136: process 1 ready
[03] 00136: process 1 running
[03] 00136: process 2 blocked (recv)
[03] 00137: process 1 blocked (send)
[03] 00137: process 2 ready
[03] 00137: process 2 running
[03] 00138: process 1 blocked
[03] 00138: process 2 blocked (send)
[03] 00139: process 2 blocked
[03] 00140: process 1 finished
[03] 00141: process 2 finished
| 00140 | Proc 03.01 | Run 60, Block 20, Wait 40, Sends 10, Recvs 10
| 00141 | Proc 01.01 | Run 30, Block 30, Wait 1, Sends 10, Recvs 10
| 00141 | Proc 02.01 | Run 50, Block 50, Wait 3, Sends 10, Recvs 10
| 00141 | Proc 03.02 | Run 60, Block 20, Wait 41, Sends 10, Recvs 10
| 00144 | Proc 01.02 | Run 30, Block 30, Wait 2, Sends 10, Recvs 10
| 00146 | Proc 02.02 | Run 50, Block 50, Wait 13, Sends 10, Recvs 10
